astropy/stats/sigma_clipping.py

```diff
@@ -452,10 +452,4 @@ def sigma_clipped_stats(data, mask=None, mask_value=None, sigma=3.0,
     median = np.ma.median(data_clip, axis=axis)
     std = np.ma.std(data_clip, ddof=std_ddof, axis=axis)
 
-    if axis is None and np.ma.isMaskedArray(median):
-        # With Numpy 1.10 np.ma.median always return a MaskedArray, even with
-        # one element. So for compatibility with previous versions, we take the
-        # scalar value
-        median = median.item()
```

Comment:
Failure comes from here. Removing this changes the behavior of `sigma_clipped_stats`, not sure if it's desirable ? ping @crawfordsm @larrybradley

Reply:
fair enough, I'll add this back with a slight modification of the comment, as I misunderstand that it talks about previous numpy versions while it's also about compatibility with previous astropy, too